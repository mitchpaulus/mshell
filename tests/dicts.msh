{
    "hello": "world",
} str wl

{
    "key": 1,
    "other key": 2,
} str wl

# Test not ending with comma
{ "key": 1, "other key": 2 3 +} str wl


{ "key": 1 2 + } dict2! @dict2 str wl

@dict2 "key" get 3 + str wl

@dict2 "key" 5 setd
@dict2 "key" get 3 + str wl

@dict2 "key"       in ("key found in dict") ("key not found in dict") iff wl
@dict2 "other key" in ("key found in dict") ("key not found in dict") iff wl

# getDef
{} "should not be found" 10 getDef str wl

# Test set leaves dict on stack
{} "key" 10 set str wl

# 'keys' and 'values'
{ "key1": 1, "key2": 2, "key3": 3 } keys str wl
{ "key1": 1, "key2": 2, "key3": 3 } values str wl

{} keys str wl
{} values str wl
