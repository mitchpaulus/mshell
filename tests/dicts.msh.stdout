is here
also here
is here
also here
-1
-2
-1
-2
{"literalKey": {"type": "String", "content": "Also should work"}}
{"hello": {"type": "String", "content": "world"}, "single quote "" string": {"type": "String", "content": "should work"}}
hello
single quote "" string
{"key": {"type": "Integer", "value": 1}, "other key": {"type": "Integer", "value": 2}}
{"key": {"type": "Integer", "value": 1}, "other key": {"type": "Integer", "value": 5}}
{"key": {"type": "Integer", "value": 3}}
6
8
key found in dict
key not found in dict
10
{"key": {"type": "Integer", "value": 10}}
["key1" "key2" "key3"]
[1 2 3]
[]
[]
1
2
